org.hl7.fhir.utilities/src/main/java/org/hl7/fhir/utilities/xml/XMLWriter.java

```diff
@@ -246,26 +246,14 @@ private int writeAttributeSet(boolean special, int col, int wrap) throws IOExcep
 					write(element[0]);
 					write("=\"");
 					if (element[1] != null)
-						write(xmlEscape(element[1]));
+						write(XMLUtil.escapeXML(element[1], charset, false));
```

Comment:
i introduced the strange duplicate behaviour hier. element[1] ist already xmlEscape'd from the setAttribute functions. however the \r and \n are not yet converted to one line. the local xmlEscape function is doing that (c < ' ') but also for the umlaute (c > '~').

Reply:
the NarrativeGenerationTests for the file actcode expects that the attributes are on one line and CR, LF are encoded in the attribute itself (e.g. same as  actcode-input.xml):

actcode-expected.xml: 

```xml
  <concept>
    <code value="_ActControlVariable"/>
    <display value="ActControlVariable"/>
    <definition value=".... snip ... pressure.&#xD;&#xA;&#xD;&#xA;Control variables ...snip..."/>
```